### ISSUE TITLE
Roundabouts - keep info until exited

### ISSUE THREAD

**Issue by Barrocas** [osmandapp/Osmand#2191]
When navigating and aproaching a roundabout, whe see (and listen the TTS voice) the top left icon info about what exit should be taken.

As soon as the roundabout is reached, that info is replaced by the next turn info. That is not good.

The displayed top left info about the roundabouts exit should be maintained visible until the exit is taken.

Worst cenario, sometimes the next turn info is another roundabout, and when inside the roundabout, the user gets wrong info, for example the map shows to take the 2nd exit, but the top left icon is already showing to take the 1st exit (next turn after exiting this one), misleading info.

**Comment by vshcherb** [comment]
I guess it is duplicated one, but any way let's keep it.

**Comment by Barrocas** [comment]
It was suggested in another issue, and it was advised to be opened as a new
issue, and so I did open this one to be a separate issue.

-- 
Cumprimentos,
D.er Hugo Barrocas

**Comment by sonora** [comment]
I am not sure I can confirm the observation. Like with any other turn, we actually keep the current turn information visible until the very point where the actual turn happens. So the question may be: When exactly did your device (via its GPS) think you reached that point. So chances are that we are not really looking at code bug in connection with roundabouts, but an issue of GPS reception and how we treat resulting location uncertainties. This would then not be limited to roundabouts, the issue may appear at other turns as well (there likely with less impact).

**Comment by Barrocas** [comment]
No, what happens is that the roundabout exit info, like 'take the second
exit' (in the top left corner icon) is removed when we enter the roundabout.
It should keep that info displayed until the exit is taken.


-- 
Cumprimentos,
D.er Hugo Barrocas

**Comment by sonora** [comment]
Apologies, you are right. We do actually combine each roundabout into one turn, and it is located at where you enter the roundabout - that's what is causing you observation.
